### PR TITLE
[NFC][AMDGPU] Pre-commit tests for buffer contents legalization

### PR DESCRIPTION
**Pull request by krzysz00**
Currently, many attempts to lower loads and stores on buffer fat pointers lower directly to intrinsic calls that will be unsupported by or crash codegen (ex, storing a [2 x i32], a <6 x half>, or an i160).

Record the current behavior to make the effects of the fix more visible in an upcoming PR.

### CODE REVIEW

File: llvm/test/CodeGen/AMDGPU/lower-buffer-fat-pointers-contents-legalization.ll [hunk 1]
```diff
@@ -0,0 +1,1685 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt -S -mcpu=gfx900 -amdgpu-lower-buffer-fat-pointers < %s | FileCheck %s
+; RUN: opt -S -mcpu=gfx900 -passes=amdgpu-lower-buffer-fat-pointers < %s | FileCheck %s
+
+target datalayout = "e-p:64:64-p1:64:64-p2:32:32-p3:32:32-p4:64:64-p5:32:32-p6:32:32-p7:160:256:256:32-p8:128:128-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-v2048:2048-n32:64-S32-A5-G1-ni:7:8"
```
Comment (shiltian) [inline llvm/test/CodeGen/AMDGPU/lower-buffer-fat-pointers-contents-legalization.ll:5]: These two are not needed if using `-mtriple=amdgcn`

**Review by shiltian — APPROVED** [review]

**Comment by shiltian** [comment]
`[NFC]`

File: llvm/test/CodeGen/AMDGPU/lower-buffer-fat-pointers-contents-legalization.ll (hunk 1 above)
Comment (krzysz00) [reply llvm/test/CodeGen/AMDGPU/lower-buffer-fat-pointers-contents-legalization.ll:5]: All the existing buffer fat pointer tests explicitly spell out the data layout so I figured I'd follow convention

Comment (arsenm) [reply llvm/test/CodeGen/AMDGPU/lower-buffer-fat-pointers-contents-legalization.ll:5]: It's rare to have datalayout / triple in the test

File: llvm/test/CodeGen/AMDGPU/lower-buffer-fat-pointers-contents-legalization.ll [hunk 2]
```diff
@@ -0,0 +1,1685 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt -S -mcpu=gfx900 -amdgpu-lower-buffer-fat-pointers < %s | FileCheck %s
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/lower-buffer-fat-pointers-contents-legalization.ll:2]: Don't bother with the old PM run line

File: llvm/test/CodeGen/AMDGPU/lower-buffer-fat-pointers-contents-legalization.ll (hunk 1 above)
Comment (krzysz00) [reply llvm/test/CodeGen/AMDGPU/lower-buffer-fat-pointers-contents-legalization.ll:5]: Should I fix that up here or in a separate bit of NFC?

(And this is testing a pass that needs the layout set correctly)

Comment (shiltian) [reply llvm/test/CodeGen/AMDGPU/lower-buffer-fat-pointers-contents-legalization.ll:5]: The default layout should be the correct one as long as the triple is set correctly. The explicit data layout specification is only required if one wants to override the default one (for whatever reason).

**Pull request merged**